ADAM version 0.19.0 will not run on Spark version 2.0.0

We knew this would happen; with Spark version 2.0.0 installed from Homebrew and ADAM version 0.19.0 installed from Homebrew Science
```bash
$ spark-submit --version
Welcome to
      ____              __
     / __/__  ___ _____/ /__
    _\ \/ _ \/ _ `/ __/  '_/
   /___/ .__/\_,_/_/ /_/\_\   version 2.0.0
      /_/
                        
Branch 
Compiled by user jenkins on 2016-07-19T21:16:09Z
Revision 
Url 
Type --help for more information.

$ adam-submit --version
Using ADAM_MAIN=org.bdgenomics.adam.cli.ADAMMain
Using SPARK_SUBMIT=/usr/local/bin/spark-submit
java.lang.NoClassDefFoundError: org/apache/spark/Logging
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:760)
	at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
	at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
	at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:368)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at org.apache.spark.util.Utils$.classForName(Utils.scala:225)
	at org.apache.spark.deploy.SparkSubmit$.org$apache$spark$deploy$SparkSubmit$$runMain(SparkSubmit.scala:686)
	at org.apache.spark.deploy.SparkSubmit$.doRunMain$1(SparkSubmit.scala:185)
	at org.apache.spark.deploy.SparkSubmit$.submit(SparkSubmit.scala:210)
	at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:124)
	at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.spark.Logging
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 19 more
```

ADAM git HEAD version 0.19.1-SNAPSHOT runs ok
```bash
$ ./bin/adam-submit --version
Using ADAM_MAIN=org.bdgenomics.adam.cli.ADAMMain
Using SPARK_SUBMIT=/usr/local/bin/spark-submit

       e         888~-_          e             e    e
      d8b        888   \        d8b           d8b  d8b
     /Y88b       888    |      /Y88b         d888bdY88b
    /  Y88b      888    |     /  Y88b       / Y88Y Y888b
   /____Y88b     888   /     /____Y88b     /   YY   Y888b
  /      Y88b    888_-~     /      Y88b   /          Y888b

ADAM version: 0.19.1-SNAPSHOT
Commit: c23aace7b1739e29935cd3f46781ae0befc0a756 Build: 2016-07-29
Built for: Scala 2.10 and Hadoop 2.6.0
```

Note the Spark binary tarball used by Homebrew was built against Scala 2.11 and Hadoop 2.7
https://www.apache.org/dyn/closer.lua?path=spark/spark-2.0.0/spark-2.0.0-bin-hadoop2.7.tgz

whereas we still build by default for Hadoop 2.6.0 and use the Scala 2.10 binary tarball for the Homebrew Science formula.

--- a/adam-core/src/main/scala/org/bdgenomics/adam/rdd/variation/VariantContextRDD.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/rdd/variation/VariantContextRDD.scala
@@ -24,8 +24,7 @@ import htsjdk.variant.variantcontext.writer.{
 import htsjdk.variant.vcf.{ VCFHeader, VCFHeaderLine }
 import java.io.OutputStream
 import org.apache.hadoop.io.LongWritable
-import org.apache.hadoop.fs.{ FileSystem, Path }
-import org.apache.spark.Logging
+import org.apache.hadoop.fs.Path
 import org.apache.spark.rdd.RDD
 import org.bdgenomics.adam.converters.{
   SupportedHeaderLines,
@@ -38,6 +37,7 @@ import org.bdgenomics.adam.models.{
 }
 import org.bdgenomics.adam.rdd.{ FileMerger, MultisampleGenomicRDD }
 import org.bdgenomics.formats.avro.Sample
+import org.bdgenomics.utils.misc.Logging
 import org.bdgenomics.utils.cli.SaveArgs
 import org.seqdoop.hadoop_bam._
 import scala.collection.JavaConversions._

--- a/pom.xml
+++ b/pom.xml
@@ -28,7 +28,7 @@
     <hadoop-bam.version>7.6.0</hadoop-bam.version>
     <slf4j.version>1.7.21</slf4j.version>
     <bdg-formats.version>0.9.0</bdg-formats.version>
-    <bdg-utils.version>0.2.7</bdg-utils.version>
+    <bdg-utils.version>0.2.8</bdg-utils.version>
     <htsjdk.version>2.5.0</htsjdk.version>
     <scoverage.plugin.version>1.1.1</scoverage.plugin.version>
   </properties>

--- a/scripts/jenkins-test
+++ b/scripts/jenkins-test
@@ -38,6 +38,13 @@ then
     exit 1
 fi
 
+# are we testing for spark 2.0.0? if so, we need to rewrite our poms first
+if [ ${SPARK_VERSION} == 2.0.0 ];
+then
+        echo "Rewriting POM.xml files for Scala 2.10."
+        ./scripts/move_to_spark_2.sh
+fi
+
 # are we testing for scala 2.11? if so, we need to rewrite our poms to 2.11 first
 if [ ${SCALAVER} == 2.11 ];
 then

--- a/scripts/move_to_spark_2.sh
+++ b/scripts/move_to_spark_2.sh
@@ -1,0 +1,8 @@
+#!/bin/bash
+
+set +x
+
+find . -name "pom.xml" -exec sed -e "/utils-/ s/_2.10/-spark2_2.10/g" \
+    -e "/adam-/ s/_2.10/-spark2_2.10/g" \
+    -e "/spark.version/ s/1.6.1/2.0.0/g" \
+    -i .spark2.bak '{}' \;

--- a/scripts/release/release.sh
+++ b/scripts/release/release.sh
@@ -38,7 +38,7 @@ mvn --batch-mode \
   release:perform
 
 if [ $? != 0 ]; then
-  echo "Releasing Scala 2.10 version failed."
+  echo "Releasing Spark 1, Scala 2.10 version failed."
   exit 1
 fi
 
@@ -50,7 +50,7 @@ zip -r adam-${release}.zip adam-${release}.jar adam-${release}.pom
 # do scala 2.11 release
 git checkout -b maint_2.11-${release} ${branch}
 ./scripts/move_to_scala_2.11.sh
-git commit -a -m "Modifying pom.xml files for 2.11 release."
+git commit -a -m "Modifying pom.xml files for Spark 1, Scala 2.11 release."
 mvn --batch-mode \
   -P distribution \
   -Dresume=false \
@@ -63,7 +63,47 @@ mvn --batch-mode \
   release:perform
 
 if [ $? != 0 ]; then
-  echo "Releasing Scala 2.11 version failed."
+  echo "Releasing Spark 1, Scala 2.11 version failed."
+  exit 1
+fi
+
+# do spark 2, scala 2.11 release
+git checkout -b maint_spark2_2.11-${release} ${branch}
+./scripts/move_to_spark2.sh
+git commit -a -m "Modifying pom.xml files for Spark 2, Scala 2.11 release."
+mvn --batch-mode \
+  -P distribution \
+  -Dresume=false \
+  -Dtag=adam-parent-spark2_2.11-${release} \
+  -DreleaseVersion=${release} \
+  -DdevelopmentVersion=${devel} \
+  -DbranchName=adam-spark2_2.11-${release} \
+  release:clean \
+  release:prepare \
+  release:perform
+
+if [ $? != 0 ]; then
+  echo "Releasing Spark 2, Scala 2.11 version failed."
+  exit 1
+fi
+
+# do spark 2, scala 2.10 release
+git checkout -b maint_spark2_2.10-${release} ${branch}
+./scripts/move_to_scala_2.10.sh
+git commit -a -m "Modifying pom.xml files for Spark 2, Scala 2.10 release."
+mvn --batch-mode \
+  -P distribution \
+  -Dresume=false \
+  -Dtag=adam-parent-spark2_2.10-${release} \
+  -DreleaseVersion=${release} \
+  -DdevelopmentVersion=${devel} \
+  -DbranchName=adam-spark2_2.10-${release} \
+  release:clean \
+  release:prepare \
+  release:perform
+
+if [ $? != 0 ]; then
+  echo "Releasing Spark 2, Scala 2.10 version failed."
   exit 1
 fi
 

